subprojects/docs/src/snippets/tasks/incrementalBuild-incrementalBuildAdvanced/groovy/build.gradle

```
@@ -50,7 +50,7 @@ tasks.register('processTemplates2', ProcessTemplates) {
 // tag::failed-inferred-task-dep[]
 
 tasks.register('badInstrumentClasses', Instrument) {
-    classFiles.from fileTree(tasks.named('compileJava').map { it.destinationDir })
+    classFiles.from fileTree(tasks.named('compileJava').map { it.destinationDirectory.get().getAsFile() })
```

This might be...

```suggestion
    classFiles.from fileTree(tasks.named('compileJava').flatMap { it.destinationDirectory })
```
Or even
```
    classFiles.from tasks.named('compileJava')
```

---

Changed.